iis/applications/client.go

```diff
@@ -0,0 +1,8 @@
+package applications
+
+import "github.com/tombuildsstuff/golang-iis/iis/cmd"
+
+// Manage IIS Applications
+type ApplicationsClient struct {
+	cmd.Client
+}
```

Comment:
could we add an acceptance test for this - for example `lifecycle_test.go` with something like this, which creates an App Pool and Website and then creates 3 x Web Applications in that to confirm this works (based on the example you've posted above):

```
package applications

import (
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"os"
	"strings"
	"testing"

	"github.com/tombuildsstuff/golang-iis/iis"
	"github.com/tombuildsstuff/golang-iis/iis/helpers"
)

func TestApplicationsLifecycle(t *testing.T) {
	rInt := helpers.RandomInt()
	appPoolName := fmt.Sprintf("app-pool-%d", rInt)
	websiteName := fmt.Sprintf("website-%d", rInt)
	rootDirectory := fmt.Sprintf("C:\\inetpub")

	log.Printf("Creating IIS Client")
	client, err := iis.NewClient()
	if err != nil {
		t.Fatalf("Error creating IIS client: %s", err)
	}

	// create the App Pool and Website to host the Web Applications in
	t.Logf("[DEBUG] Creating App Pool %q..", appPoolName)
	if err := client.AppPools.Create(appPoolName); err != nil {
		t.Fatalf("Error creating App Pool %q: %s", appPoolName, err)
	}
	mainWebsitePath := fmt.Sprintf("%s\\wwwroot", rootDirectory)
	t.Logf("[DEBUG] Creating Website at %q..", mainWebsitePath)
	if err := client.Websites.Create(websiteName, appPoolName, mainWebsitePath); err != nil {
		t.Fatalf("Error creating Website %q (App Pool %q): %s", websiteName, appPoolName, err)
	}

	// then let's create 3 Web Applications
	applicationsCreated := make([]int, 0)
	for i := 1; i <= 3; i++ {
		name := fmt.Sprintf("app-%d-%d", rInt, i)
		testAppPath := fmt.Sprintf("%s\\%s", rootDirectory, name)

		// first create the content
		t.Log("[DEBUG] Creating the Test Directory")
		err := createTestDirectory(t, testAppPath, testAppContent(i))
		if err != nil {
			t.Fatal(err)
		}

		// so we can clean these up later on
		applicationsCreated = append(applicationsCreated, i)

		// then attach it to the site
		t.Logf("[DEBUG] Creating Web App %q..", name)
		if err := client.Applications.Create(name, websiteName, appPoolName, testAppPath); err != nil {
			t.Fatalf("Error creating Web Application %q (Website %q / App Pool %q): %s", name, websiteName, appPoolName)
		}
	}

	// assert things exist
	for _, i := range applicationsCreated {
		testUrl := fmt.Sprintf("http://localhost/app-%d-%d", rInt, i)
		name := fmt.Sprintf("app-%d-%d", rInt, i)
		testAppPath := fmt.Sprintf("%s\\%s", rootDirectory, name)

		t.Logf("[DEBUG] Asserting Web Application %d exists in IIS..", i)
		exists, err := client.Applications.Exists(name, websiteName)
		if err != nil {
			t.Fatalf("Error determining if Application %q exists: %s", name, err)
		}
		if !*exists {
			t.Fatalf("Expected Application %q to exist but it didn't!", name)
		}

		t.Logf("[DEBUG] Reading Properties for Web Application %d from IIS..", i)
		props, err := client.Applications.Get(name, websiteName)
		if err != nil {
			t.Fatalf("Error retrieving Application %q (Website %q): %s", name, websiteName, err)
		}

		if !strings.EqualFold(props.PhysicalPath, testAppPath) {
			t.Fatalf("Expected Application to be at %q but got %q", testAppPath, props.PhysicalPath)
		}

		t.Logf("[DEBUG] Retrieving Application %q from %q..", name, testUrl)
		resp, err := http.Get(testUrl)
		if err != nil {
			t.Fatalf("Error trying to get the test HTML from IIS: %s", err)
		}
		respContent, err := ioutil.ReadAll(resp.Body)
		if err != nil {
			t.Fatalf("Error getting HTTP response body for : %s", err)
		}

		expectedBody := testAppContent(i)
		actualBody := string(respContent)
		if expectedBody != actualBody {
			t.Fatalf("[DEBUG] Expected %q but got %q", expectedBody, actualBody)
		}
	}

	// then we need to tear things down in order
	for _, i := range applicationsCreated {
		t.Logf("[DEBUG] Removing Web Application %d..", i)
		name := fmt.Sprintf("app-%d-%d", rInt, i)
		testAppPath := fmt.Sprintf("%s\\%s", rootDirectory, name)

		t.Logf("[DEBUG] Deleting App from IIS..")
		if err := client.Applications.Delete(name, websiteName); err != nil {
			t.Fatalf("Error deleting Application %q (Website %q)..", name, websiteName)
		}

		// cleanup the content
		t.Logf("[DEBUG] Deleting App from File System..")
		if err := os.RemoveAll(testAppPath); err != nil {
			t.Fatalf("Error cleaning up the test files for Application %q (Website %q): %s", name, websiteName, err)
		}
	}

	// then remove the website/app pool
	t.Logf("[DEBUG] Deleting Website..")
	if err := client.Websites.Delete(websiteName); err != nil {
		t.Fatalf("Error deleting Website %q (App Pool %q): %s", websiteName, appPoolName, err)
	}

	t.Logf("[DEBUG] Deleting App Pool..")
	if err := client.AppPools.Delete(appPoolName); err != nil {
		t.Fatalf("Error deleting App Pool %q: %s", appPoolName, err)
	}
}

func createTestDirectory(t *testing.T, testAppPath string, testAppContent string) error {
	t.Logf("[DEBUG] Creating test application folder %q", testAppPath)
	err := os.MkdirAll(testAppPath, os.ModeDir)
	if err != nil {
		return fmt.Errorf("Error creating directory %q: %s", testAppPath, err)
	}

	t.Logf("[DEBUG] Writing test HTML file")
	testFile, err := os.OpenFile(fmt.Sprintf("%s\\index.html", testAppPath), os.O_CREATE|os.O_WRONLY, os.ModePerm)
	defer testFile.Close()

	_, err = testFile.WriteString(testAppContent)
	if err != nil {
		return fmt.Errorf("Error writing test HTML file: %s", err)
	}

	t.Logf("[DEBUG] Wrote test HTML content: %q", testAppContent)

	return nil
}

func testAppContent(rInt int) string {
	return fmt.Sprintf(`<html><head><title>IIS Testing</title></head><body><h1>Hello World from Application %d!</h1></body></html>`, rInt)
}
```